src/settings_internal.js

```diff
@@ -209,3 +209,6 @@ var STACK_BASE = 0;
 // enabled, but if `--no-entry` is passed, or if `_main` is not part of 
 // EXPORTED_FUNCTIONS then this gets set to 0.
 var EXPECT_MAIN = 1;
+
+// Provide a Promise as a result from the MODULARIZE factory function.
+var MODULARIZE_PROMISE = 1;
```

Comment:
Just thinking forward, ideally the ready promise should be available for non-MODULARIZE builds too. Most of the promise code is probably compatible with that already. So rather than calling this setting MODULARIZE_PROMISE, maybe call it something like EXPORT_READY_PROMISE (so that it doesn't have to be renamed later)?

Reply:
Thanks @curiousdannii , good idea! Done.